scripts/update_todo_references.py

```diff
@@ -0,0 +1,248 @@
+import json
+import os
+import re
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+PRIORITIES = {"LOW": 4, "MEDIUM": 3, "HIGH": 2}
+
+JIRA_BASE_URL = os.environ["JIRA_BASE_URL"]
+JIRA_USERNAME = os.environ["JIRA_USERNAME"]
+JIRA_API_TOKEN = os.environ["JIRA_API_TOKEN"]
+
+JIRA_BOARD_ID = os.environ["JIRA_BOARD_ID"]
+JIRA_ISSUE_TYPE_ID = os.environ["JIRA_ISSUE_TYPE_ID"]
+JIRA_DONE_STATUS_CATEGORY_ID = 3
+JIRA_ISSUE_LABEL = "todo_comment"
+
+
+class Todo:
+    """Represents a TODO comment in code."""
+    def __init__(
+        self,
+        filepath: str,
+        description: str,
+        priority: Optional[str] = None,
+        issue_key: Optional[str] = None,
+    ):
+        self.filepath = filepath
+        self.description = description
+        self.priority = priority
+        self.issue_key = issue_key
+
+
+class JiraIssue:
+    """Represents an issue from a JIRA board."""
+    def __init__(self, key: str, done: bool):
+        self.key = key
+        self.done = done
+
+
+def issue_payload_for_todo(todo: Todo) -> Dict[str, Any]:
+    """Returns the http request data required to create a JIRA issue for the
+    given Todo object."""
+    title = (
+        f"Autogenerated issue from TODO comment in file {todo.filepath}."
+    )
+    return {
+        "fields": {
+            "summary": title,
+            "issuetype": {"id": f"{JIRA_ISSUE_TYPE_ID}"},
+            "project": {"id": f"{JIRA_BOARD_ID}"},
+            "priority": {"id": f"{PRIORITIES[todo.priority]}"},
+            "labels": [JIRA_ISSUE_LABEL],
+            "description": {
+                "type": "doc",
+                "version": 1,
+                "content": [
+                    {
+                        "type": "paragraph",
+                        "content": [{"text": todo.description, "type": "text"}],
+                    }
+                ],
+            },
+        },
+    }
+
+
+def create_jira_issues(todos: List[Todo]) -> None:
```

Comment:
There is no error handling in this function. I would suggest to break into two functions and:

* Unit test them
* Add error catching and retry with backoffs?

Reply:
Where do you think we should put the tests? The tests directory seems a little strange for a dev script like this

Reply:
We can put it in `tests/scripts/` I think that makes sense

Reply:
Alright I added some tests for it there